azure-key-vault-properties-provider/2.1/modules/ROOT/pages/azure-key-vault-properties-provider-reference.adoc

```diff
@@ -0,0 +1,89 @@
+= Azure Key Vault Properties Provider 2.1 Reference - Mule 4
+
+Azure Key Vault Properties Provider is a configuration that enables you to retrieve your secrets from a secure Azure cloud service before your Mule app runs.
+
+[[Config]]
+== Configuration
+
+Default Configuration for Azure Key Vault Properties Provider.
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Name | Type | Description | Default Value | Required
+|Name | String | Name for this configuration. Connectors reference the configuration with this name. | | x
+| Client Id  | String | Client ID of your app. If not set, the properties provider uses the Azure SDK default authentication and looks for this value in the environment variable `AZURE_CLIENT_ID`. Note: `tenantId`, `clientId`, and `clientSecret` must all be set together if specified in the configuration. | | 
+| Tenant Id  | String | Tenant ID of your app. If not set, the properties provider uses the Azure SDK default authentication and looks for this value in the environment variable `AZURE_TENANT_ID`. Note: `tenantId`, `clientId`, and `clientSecret` must all be set together if specified in the configuration. | | 
+| Client Secret | String | Client secret of your app. If not set, the properties provider uses the Azure SDK default authentication and look for this value in the environment variable `AZURE_CLIENT_SECRET`. Note: `tenantId`, `clientId`, and `clientSecret` must all be set together if specified in the configuration. | | 
+| Key Vault Name | String | Name of the vault that you want to connect. If not set, the properties provider uses the system property `MULE_AZURE_KEY_VAULT_NAME`. | | 
+| Uri Override | String |  The URI override for the key vault. |  | 
+|===
+
+== Requirements
+
+You must add the following repository and its associated credentials in the Maven settings:
+
+[source,xml,linenums]
+----
+<repository>
+          <id>MuleRepository2</id>
+          <name>MuleRepository2</name>
+          <url>https://repository.mulesoft.org/nexus/content/repositories/releases-ee/</url>
+          <layout>default</layout>
+          <releases>
+            <enabled>true</enabled>
+          </releases>
+          <snapshots>
+            <enabled>true</enabled>
+          </snapshots>
+</repository>
+----
+== Limitations
+
+Azure Key Vault Properties Provider doesn't support the use of property files for the configuration of the properties provider. 
+
+== Environment Variables
+
+You can use the following environment variables to configure the behavior of the Azure Key Vault Properties Provider from the environment instead of specifying parameters in the Mule app configuration:
```

Comment:
```suggestion
Use the following environment variables to configure the behavior of the Azure Key Vault Properties Provider from the environment instead of specifying parameters in the Mule app configuration:
```